framework/src/base/MooseApp.C

```diff
@@ -279,6 +298,64 @@ MooseApp::MooseApp(InputParameters parameters)
 
   if (_check_input && isParamValid("recover"))
     mooseError("Cannot run --check-input with --recover. Recover files might not exist");
+
+  if (isParamValid("start_in_debugger"))
+  {
+    auto command = getParam<std::string>("start_in_debugger");
+
+    Moose::out << "Starting in debugger using: " << command << std::endl;
+
+    auto hostname = MooseUtils::hostname();
+
+    std::stringstream command_stream;
+
+    // This will start XTerm and print out some info first... then run the debugger
+    command_stream << "xterm -e \"echo 'Rank: " << processor_id() << "  Hostname: " << hostname
+                   << "  PID: " << getpid() << "'; echo ''; ";
+
+    // Figure out how to run the debugger
+    if (command.find("lldb") != std::string::npos || command.find("gdb") != std::string::npos)
+      command_stream << command << " -p " << getpid();
+    else
+      mooseError("Unknown debugger: ",
+                 command,
+                 "\nIf this is truly what you meant then contact moose-users to have a discussion "
+                 "about adding your debugger.");
```

Comment:
Or add a wrapper script that contains the strings `gdb` or `lldb` :-D. This is a bit... hacky.

Reply:
This is not at all hacky - it's a good first attempt.  I don't have time to go through every compiler in the world - but in the case where you run something unexpected I wanted to provide some meaningful output.

What's "hacky" about it?  You think it should just say "unknown debugger" and quit?

Reply:
This looks fine to me.

Reply:
I want to point out that the reason I searched for `lldb` and `gdb` is because you might need to run a compound statement (like `sudo lldb`).

I don't see a better way to test for `lldb`/`gdb`... but I'm open to ideas.